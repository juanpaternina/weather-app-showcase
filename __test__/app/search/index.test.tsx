import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import Search from '@/app/search';

import { useRouter } from 'expo-router';

jest.mock('expo-router', () => ({
  useRouter: jest.fn(),
}));

jest.mock('@expo/vector-icons/MaterialIcons', () => ({
  __esModule: true,
  default: ({ name, size, color }) => (
    <mock-icon test-id={`material-icon-${name}`} size={size} color={color} />
  ),
}));

jest.mock('@/components/SearchField', () => ({
  SearchField: jest.fn(() => <></>),
}));

describe('Search Component', () => {
  beforeAll(() => {
    jest.clearAllMocks();
    (useRouter as unknown as jest.Mock).mockReturnValue({
      back: jest.fn(),
    });
  });

  test('It should render SearchField component', () => {
    const { toJSON } = render(<Search />);
    expect(toJSON()).toMatchSnapshot();
  });

  test('Back button should navigate to the previous screen', () => {
    const { getByTestId } = render(<Search />);
    const backButton = getByTestId('back-button');
    fireEvent.press(backButton);

    expect(useRouter().back).toHaveBeenCalledTimes(1);
  });
});
