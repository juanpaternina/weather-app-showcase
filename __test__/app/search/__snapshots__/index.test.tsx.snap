// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component It should render SearchField component 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "white",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "white",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingLeft": 20,
        "paddingRight": 20,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "marginBottom": 7,
        }
      }
      testID="back-button"
    >
      <mock-icon
        color="black"
        size={32}
        test-id="material-icon-arrow-back"
      />
    </View>
  </View>
</RCTSafeAreaView>
`;
